lib/dependency-manager-adapters/npm.js

```diff
@@ -124,15 +124,43 @@ module.exports = CoreObject.extend({
     fs.writeFileSync(packageJSONFile, JSON.stringify(newPackageJSON, null, 2));
   },
   _packageJSONForDependencySet(packageJSON, depSet) {
-
     this._overridePackageJSONDependencies(packageJSON, depSet, 'dependencies');
     this._overridePackageJSONDependencies(packageJSON, depSet, 'devDependencies');
     this._overridePackageJSONDependencies(packageJSON, depSet, 'peerDependencies');
 
+    if (this.useYarnCommand) {
+      this._addYarnResolutions(packageJSON, depSet);
+    }
+
     return packageJSON;
   },
+
+  _addYarnResolutions(packageJSON, depSet) {
```

Comment:
I made this its own function. I was worried about littering `_overridePackageJSONDependencies` with a bunch of `if (this.useYarnCommand)` guards.